plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -49,53 +50,60 @@ export class PersonState {
     private personManager: PersonManager
 
     constructor(
+        event: PluginEvent,
         teamId: number,
         distinctId: string,
-        eventProperties: Properties,
         timestamp: DateTime,
         db: DB,
         statsd: StatsD | undefined,
-        personManager: PersonManager
+        personManager: PersonManager,
+        uuid?: UUIDT
     ) {
-        this.teamId = teamId
+        this.event = event
         this.distinctId = distinctId
-        this.eventProperties = eventProperties
+        this.teamId = teamId
+        this.eventProperties = event.properties!
         this.timestamp = timestamp
-        this.newUuid = new UUIDT().toString()
+        this.newUuid = (uuid || new UUIDT()).toString()
 
         this.db = db
         this.statsd = statsd
         this.personManager = personManager
     }
 
-    async updateProperties(): Promise<void> {
-        const wasPersonCreated = await this.createPersonIfDistinctIdIsNew()
+    async update(): Promise<Person | undefined> {
+        await this.handleIdentifyOrAlias()
+        return await this.updateProperties()
+    }
+
+    async updateProperties(): Promise<Person | undefined> {
```

Comment:
It seems that `updateProperties()` and `handleIdentifyOrAlias()` can now be made private, as `update()` should be used instead.